Sources/Display.php

```diff
@@ -1013,13 +1018,15 @@ function Display()
 
 		// Sort the messages into the correct display order
 		if (!$DBascending)
-			sort($messages);
+			sort($messages, SORT_NUMERIC);
+		else
+			rsort($messages, SORT_NUMERIC);
 	}
 
 	// Remember the paging data for next time
-	$_SESSION['page_first_id'] = $ascending ? reset($messages) : end($messages);
+	$_SESSION['page_first_id'] = $messages[0];
 	$_SESSION['page_before_start'] = $_REQUEST['start'] - $limit;
-	$_SESSION['page_last_id'] = $ascending ? end($messages) : reset($messages);
+	$_SESSION['page_last_id'] = $messages[count($messages) - 1];
```

Comment:
Instead of `$messages[count($messages) - 1]`, you should use `end($messages)` here. If `$messages` is an empty array, `$messages[count($messages) - 1]` will evaluate to `$messages[-1]` and produce an undefined index error, whereas `reset($messages)` will simply return `false`.